### PR TITLE
feat(Server): server now accepts Fn not fn

### LINKED ISSUE #81: Move from `fn` to `Fn`
We originally used `fn` because there was no need, but building more complex stuff needs `Fn`; for example, a framework writing their own `listen` method will want to use stuff from `self` inside of the server, but you can't create that stuff inside the `fn` passed to `new`.

### PR DESCRIPTION
**Pull request by ashleygwilliams**
fixes #81

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -47,17 +47,23 @@ mod error;
...
 /// server and listen for connections.
 pub struct Server {
-    handler: fn(Request<&[u8]>, ResponseBuilder) -> Result<Response<&[u8]>, Error>,
+    handler: Handler,
 }
 
-
 impl Server {
-    /// Constructs a new server with the given handler.
+    /// Constructs a new server with the given handler
```
Comment (steveklabnik) [inline src/lib.rs]: we should revert this change, not sure how it snuck in

File: src/lib.rs [hunk 2]
```diff
@@ -84,10 +90,15 @@ impl Server {
     ///     });
     /// }
     /// ```
-    pub fn new(
-        handler: fn(Request<&[u8]>, ResponseBuilder) -> Result<Response<&[u8]>, Error>,
+    pub fn new<
+        H: Fn(Request<&[u8]>, ResponseBuilder) -> Result<Response<&[u8]>, Error>
```
Comment (gsquire) [inline src/lib.rs]: In the definition of the `Handler` type could you omit the `Box` and use that here? IMHO it reads better this way.

Comment (steveklabnik) [reply src/lib.rs]: type aliases don't work with traits, so you can't.

Comment (gsquire) [reply src/lib.rs]: Ah, good point.

Comment (ashleygwilliams) [reply src/lib.rs]: heh, we went through all of that as we paired on this today. good thought! but yeah, alas.

**Comment by hmvp** [comment]
Will this also be published to crates.io? It seems the last release is older than this..

**Comment by steveklabnik** [comment]
Yes, I have been away from simpleserver for a bit. I will try to release soon.
